src/App.js

```diff
@@ -45,21 +78,82 @@ const App = () => {
     //    3 squares in each column match
     // 3. Go across each diagonal to see if
     //    all three squares have the same value.
+
+  const checkForWinner = () => {
+    let i = 0;
+    while (i < 3) {
+      if (
+        squares[i][0].value !== '' &&
+        squares[i][0].value === squares[i][1].value &&
+        squares[i][1].value === squares[i][2].value
+      ) {
+        return squares[i][0].value;
+      } else if (
+        squares[0][i].value !== '' &&
+        squares[0][i].value === squares[1][i].value &&
+        squares[1][i].value === squares[2][i].value
+      ) {
+        return squares[0][i].value;
+      }
+      i += 1;
+    }
+    
+    // top-Left bottom-right
+    if (
+      squares[1][1].value !== '' &&
+      squares[0][0].value === squares[1][1].value &&
+      squares[1][1].value === squares[2][2].value
+    ) {
+      return squares[0][0].value;
+    }
+    
+    //top-right bottom-left
+    if (
+      squares[1][1].value !== '' &&
+      squares[0][2].value === squares[1][1].value &&
+      squares[1][1].value === squares[2][0].value
+    ) {
+      return squares[0][2].value;
+    }
+    return null; //no winner 
+  };
+
+const isBoardFull = (squares) => {
+  let count = 0;
+  let row = 0;
+  let column = 0;
+
+  while(row < 3){
+    while(column < 3){
+      if(squares[row][column] === 'X' || squares[row][column] === 'O')
+        count += 1;
+    }
+  }          
+  if(count == 9){
+      return true;
+  }
+  else{
+      return false;
+  }
```

Comment:
Couple of things:

1.  Use `===` instead of `==` 
2. Since you return `true` if the condition is `true` and `false` otherwise. 

```suggestion
  return count === 9
```